modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Tabs/Advanced/ReadOnlyContainer.tsx

```diff
@@ -72,6 +103,52 @@ export function ReadOnlyContainer({setValues, values}: ReadOnlyContainerProps) {
 					value="conditional"
 				/>
 			</ClayRadioGroup>
+
+			{readOnlySetting?.value === 'conditional' && (
+				<ExpressionBuilder
+					feedbackMessage={Liferay.Language.get(
+						'use-expressions-to-create-a-condition'
+					)}
+					label={Liferay.Language.get('expression-builder')}
+					onChange={({target: {value}}) => {
+						setValues({
+							objectFieldSettings: updateReadOnlyScriptSetting(
+								values.objectFieldSettings as ObjectFieldSetting[],
+								value
+							),
+						});
+					}}
+					onOpenModal={() => {
+						const parentWindow = Liferay.Util.getOpener();
+
+						parentWindow.Liferay.fire(
+							'openExpressionBuilderModal',
+							{
+								header: Liferay.Language.get('formula-builder'),
+								onSave: (script: string) => {
+									setValues({
+										objectFieldSettings: updateReadOnlyScriptSetting(
+											values.objectFieldSettings as ObjectFieldSetting[],
+											script
+										),
+									});
+								},
+								placeholder: `<#-- ${Liferay.Util.sub(
+									Liferay.Language.get(
+										'add-formulas-to-calculate-values-based-on-other-fields-type-x-to-use-the-autocomplete-feature'
```

Comment:
@danvb112 We cannot use this sentence because we're not providing the autocomplete feature.

Reply:
Thanks @naraandrade, I spoke with Victor and he changed the language key